### ISSUE TITLE
py27_native.test.unit._metadata_code_details_test.MetadataCodeDetailsTest segfaults on macos

### ISSUE THREAD

**Issue by markdroth** [grpc/grpc#17568]
https://source.cloud.google.com/results/invocations/e0199b9c-876b-4b4a-a76f-6800b39d3f5e/targets/github%2Fgrpc%2Fpython_macos_opt_native/tests

```
++ dirname /Volumes/BuildData/tmpfs/src/github/grpc/workspace_python_macos_opt_native/tools/run_tests/helper_scripts/run_python.sh
+ cd /Volumes/BuildData/tmpfs/src/github/grpc/workspace_python_macos_opt_native/tools/run_tests/helper_scripts/../../..
++ realpath py27_native/bin/python
+ PYTHON=/Volumes/BuildData/tmpfs/src/github/grpc/workspace_python_macos_opt_native/py27_native/bin/python
++ pwd
+ ROOT=/Volumes/BuildData/tmpfs/src/github/grpc/workspace_python_macos_opt_native
+ /Volumes/BuildData/tmpfs/src/github/grpc/workspace_python_macos_opt_native/py27_native/bin/python /Volumes/BuildData/tmpfs/src/github/grpc/workspace_python_macos_opt_native/src/python/grpcio_tests/setup.py test_lite
running test_lite
Running       unit._metadata_code_details_test.MetadataCodeDetailsTest.testAbortedStreamStream
[95mTesting gRPC Python...
[0m[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testAbortedStreamStream
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testAbortedStreamUnary
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testAbortedStreamUnary
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testAbortedUnaryStream
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testAbortedUnaryStream
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testAbortedUnaryUnary
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testAbortedUnaryUnary
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeExceptionStreamStream
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeExceptionStreamStream
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeExceptionStreamUnary
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeExceptionStreamUnary
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeExceptionUnaryStream
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeExceptionUnaryStream
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeExceptionUnaryUnary
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeExceptionUnaryUnary
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeReturnNoneStreamUnary
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeReturnNoneStreamUnary
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeReturnNoneUnaryUnary
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeReturnNoneUnaryUnary
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeStreamStream
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeStreamStream
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeStreamUnary
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeStreamUnary
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeUnaryStream
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeUnaryStream
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeUnaryUnary
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testCustomCodeUnaryUnary
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testSuccessfulStreamStream
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testSuccessfulStreamStream
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testSuccessfulStreamUnary
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testSuccessfulStreamUnary
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testSuccessfulUnaryStream
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testSuccessfulUnaryStream
[0mRunning       unit._metadata_code_details_test.MetadataCodeDetailsTest.testSuccessfulUnaryUnary
[92mSUCCESS       unit._metadata_code_details_test.MetadataCodeDetailsTest.testSuccessfulUnaryUnary
[0m18 tests finished:
	18 successful
	0 unsuccessful
	0 skipped
	0 expected failures
	0 unexpected successes
Interrupted Tests:
	[]
Errors/Failures:
Unexpected successes: []
/Volumes/BuildData/tmpfs/src/github/grpc/workspace_python_macos_opt_native/tools/run_tests/helper_scripts/run_python.sh: line 25: 80932 Segmentation fault: 11  $PYTHON "$ROOT/src/python/grpcio_tests/setup.py" "$2"
```

**Comment by ericgribkoff** [comment]
Probably the same issue as in https://github.com/grpc/grpc/issues/15308; investigating.

**Comment by gnossen** [comment]
@ericgribkoff Is this issue still valid?

**Comment by ericgribkoff** [comment]
No, this seems to have been resolved by #17579
